### ISSUE TITLE
Bug: [no-floating-promises] Does not enforce promise handling when reference is introduced

### ISSUE THREAD

**Issue by frec-nick** [typescript-eslint/typescript-eslint#5749]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=GYVwdgxgLglg9mABFApgZygCgJSIN4BQiiECGiAhogLyIAKATnALYxooB0D6cANgG4ocRRNyggGSTGBC9e1BRVwB+RAEZEALkQAmANwEAvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooiaBOiRwYAL4h1QA&tsconfig=N4XyA

### Repro Code

```typescript
function test() {
  const a = Promise.resolve()
  return (null===a) ? 1 : 2;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  {
    "rules": {
      "@typescript-eslint/no-floating-promises": "error"
    }
  }
};
```

### tsconfig

```jsonc
{}
```

### Expected Result

I expected an error to be reported on line 2 that promise `a` is never handled

### Actual Result

There is no error on line 2

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.10.0` |
| `@typescript-eslint/parser`            | `5.38.1` |
| `TypeScript`                           | `4.8.4` |
| `ESLint`                               | `8.7.0` |
| `node`                                 | `16.17.0` |

This was also tested on `typescript-estlint` SHA `7a8a0a3c500ca726d2ab3bee0ae9f3fb9d8d39b8` in `typescript-eslint/packages/eslint-plugin/tests/rules/no-floating-promises.test.ts`:
<img width="412" alt="image" src="https://user-images.githubusercontent.com/111302153/194124486-bf72e9ed-eafb-4a8c-96c0-04785cdf98b7.png">

**Comment by bradzacher** [comment]
This is working as intended.
The lint rule only looks for floating promises - that is to say expression statements that have a promise type.

If you store the promise in a variable or pass it to a function then it's no longer "floating", and it is assumed that you will "correctly" use the stored value.

Misuse of a variable can be detected separately with rules like:
- no-unused-vars
- no-misused-promises
- no-unnecessary-condition

[example](https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=GYVwdgxgLglg9mABFApgZygCgJSIN4BQiiECGiAhogLyIAKATnALYxooB0D6cANgG4ocRRNyggGSTGBC9e1BRVwB+RAEZEALkQAmANwEAvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTooiaBOiQANOGx4ipZBWp0GzNp1qwmTRGRTJ+0QrTIcmAE0pCn8yIuWQNAXxA-IA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A)
